policyengine_us/variables/gov/states/hi/tax/income/exemptions/hi_exemptions.py

```diff
@@ -0,0 +1,41 @@
+from policyengine_us.model_api import *
+
+
+class hi_exemptions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Hawaii exemptions"
+    unit = USD
+    documentation = (
+        "https://files.hawaii.gov/tax/forms/2022/n11ins.pdf#page=20"
+    )
+    definition_period = YEAR
+    defined_for = StateCode.HI
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.hi.tax.income.exemptions
+        exemp = tax_unit("exemptions", period)
+
+        disabled_head = tax_unit("head_is_disabled", period).astype(int)
+        disabled_spouse = tax_unit("spouse_is_disabled", period).astype(int)
+
+        aged_head = (tax_unit("age_head", period) >= p.age_threshold).astype(
+            int
+        )
+        aged_spouse = (
+            tax_unit("age_spouse", period) >= p.age_threshold
+        ).astype(int)
+
+        adjusted_exemp = where(
```

Comment:
Can you leave some comments explaining this logic and tying it to the form or law?

Reply:
Sorry for the late response. Here the basic logic is from the instruction for exemption calculation as below:
![image](https://github.com/PolicyEngine/policyengine-us/assets/97819588/b2fecc89-12c6-4c81-89c2-c8a22624d5cf)

https://files.hawaii.gov/tax/forms/2022/n11ins.pdf#page=20

Since if I claim the disability exemption, I won't be able to claim additional exemptions including aged exemption, so it will be excluded from the total exemptions.
The adjusted_exemp first consider the disability of the head, and consider the disability of spouse to get the exemp_base, which includes all other exemptions except disability exemptions. Hope I made it through to you and please leave any comments if there's anything that I could help to explain. Thank you!

Reply:
Sorry for the delay here. How does this look? I think avoiding the chained conditional subtractions reads more clearly while reflecting the legal code, which states:
>No additional exemptions shall be taken for dependents; the special exemption is available in lieu of any personal exemptions to which the taxpayer may be entitled. 
```suggestion
        # Head can claim the disabled exemption if they are disabled.
        head_non_disabled_exemption = p.base + p.aged * aged_head
        head_disabled_exemption = p.disabled * disabled_head
        head_exemption = max_(head_non_disabled_exemption, head_disabled_exemption)
        # Same for spouse.
        spouse_non_disabled_exemption = p.base + p.aged * aged_spouse
        spouse_disabled_exemption = p.disabled * disabled_spouse
        spouse_exemption = max_(spouse_non_disabled_exemption, spouse_disabled_exemption)
        # Add dependent exemptions only if neither head nor spouse claims disabled exemptions.
        dependent_exemptions = tax_unit("tax_unit_dependents", period) * p.base
        # Check if using disabled exemptions increases total exemption value.
        exemptions_if_claiming_disabled = head_exemption + spouse_exemption + dependent_exemptions
        exemptions_if_not_claiming_disabled = head_non_disabled_exemption + spouse_non_disabled_exemption + dependent_exemptions
        return max_(exemptions_if_claiming_disabled, exemptions_if_not_claiming_disabled)
```